socorro/processor/signature_utilities.py

```diff
@@ -683,6 +685,10 @@ def _create_frame_list(
         return frame_signatures_list
 
     #--------------------------------------------------------------------------
+    def _get_crashing_thread(self, processed_crash):
+        return processed_crash.json_dump['crash_info']['crashing_thread']
```

Comment:
such care is not necessary here.  not having the key is a known condition for the signature generation code.  It is expecting a KeyError to trigger the fallback behavior signature generation behavior.